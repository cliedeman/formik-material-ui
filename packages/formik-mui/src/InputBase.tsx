import * as React from 'react';
import { 
  type InputBaseProps as MuiInputBaseProps,
  InputBase as MuiInputBase,
} from '@mui/material';
import type { FieldProps } from 'formik';

export interface InputBaseProps
  extends FieldProps,
    Omit<MuiInputBaseProps, 'name' | 'value' | 'error'> {}

export function fieldToInputBase({
  disabled,
  field: { onBlur: fieldOnBlur, ...field },
  form: { isSubmitting },
  onBlur,
  ...props
}: InputBaseProps): MuiInputBaseProps {
  return {
    disabled: disabled ?? isSubmitting,
    onBlur:
      onBlur ??
      function (e) {
        fieldOnBlur(e ?? field.name);
      },
    ...field,
    ...props,
  };
}

export function InputBase(props: InputBaseProps) {
  return <MuiInputBase {...fieldToInputBase(props)} />;
}

InputBase.displayName = 'FormikMaterialUIInputBase';
