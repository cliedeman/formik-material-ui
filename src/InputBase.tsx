import * as React from 'react';
import MuiInputBase, {
  InputBaseProps as MuiInputBaseProps,
} from '@material-ui/core/InputBase';
import { FieldProps } from 'formik';

export interface InputBaseProps
  extends FieldProps,
    Omit<MuiInputBaseProps, 'name' | 'onChange' | 'value'> {}

export const fieldToInputBase = ({
  field,
  form: { isSubmitting },
  disabled,
  ...props
}: InputBaseProps): MuiInputBaseProps => {
  return {
    disabled: disabled ?? isSubmitting,
    ...props,
    ...field,
  };
};

export const InputBase: React.ComponentType<InputBaseProps> = (
  props: InputBaseProps
) => <MuiInputBase {...fieldToInputBase(props)} />;

InputBase.displayName = 'FormikMaterialUIInputBase';
