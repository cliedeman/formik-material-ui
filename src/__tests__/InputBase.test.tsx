import * as React from 'react';
import { Formik, Field, Form } from 'formik';
import renderer from 'react-test-renderer';

import { InputBase } from '../InputBase';

test('InputBase Renders Correctly', () => {
  const component = renderer.create(
    <Formik onSubmit={() => {}} initialValues={{ test: 'Input' }}>
      <Form>
        <Field name="test" label="InputBase" component={InputBase} />
      </Form>
    </Formik>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
