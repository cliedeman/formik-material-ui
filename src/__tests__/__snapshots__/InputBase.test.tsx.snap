// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBase Renders Correctly 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiInputBase-root"
    label="InputBase"
    onClick={[Function]}
  >
    <input
      className="MuiInputBase-input"
      disabled={false}
      name="test"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="text"
      value="Input"
    />
  </div>
</form>
`;
